<launch>
      <include file="$(find turtlebot_gazebo)/launch/turtlebot_world.launch"/>
      <node name="mux" pkg="ros_turtlebot_pilot" type="mux" output="screen"/>
      <node name="joy_cmd" pkg="ros_turtlebot_pilot" type="joy_cmd" output="screen"/>
      <node name="controller" pkg="ros_turtlebot_pilot" type="controller" output="screen">
      <param name = "odom" value="improved_odom"/> <!--possible values: odom, improved_odom"-->
      </node>
      <!-- PID -->
      <node name="PID_Controller" pkg="pid" type="controller">
            <param name="Kp" value="1" />
            <param name="Ki" value="0.0" />
            <param name="Kd" value="0.5" />
            <param name="upper_limit" value="1" />
            <param name="lower_limit" value="-10" />
            <param name="windup_limit" value="10" />
            <param name="diagnostic_period" value="0.25" />
            <param name="max_loop_frequency" value="100.0" />
            <param name="min_loop_frequency" value="100.0" />
      </node>
      <node name="improved_odometry" pkg="ros_turtlebot_pilot" type="improved_odometry" output="screen">
            <param name="odom_method" value="exact"/><!--possible values: euler, runge-kutta, exact-->
            <param name="data_num" value="1"/> <!--number of data read that are then averaged-->
      </node>
</launch>
